blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -48,13 +50,18 @@ object BlazeClient {
               .flatMap { _ => manager.release(connection) }
             Task.now(DisposableResponse(r, dispose))
 
-          case -\/(Command.EOF) =>
+          case -\/(Command.EOF) if !isFresh =>
             invalidate(connection).flatMap { _ =>
-              manager.borrow(key).flatMap { newConn =>
-                loop(newConn, flushPrelude)
+              manager.borrow(key, true).flatMap { newConn =>
+                loop(newConn, flushPrelude, true)
```

Comment:
If we call `borrow(key, true)` then since we required a new connection to be created in `borrow`, we know that the `isFresh` param in `loop` will also need to be set to `true`.

But if we call `borrow(key, false)` then isn't it still possible that `borrow` creates a new connection? Would it make sense for `borrow` to return `Task[(Boolean, A)] ` where the `Boolean` signifies if a new connection was created, so we can then `flatMap` into that and be provided the correct `Boolean` value to pass to `loop` without having to provide it ourselves?

I think if we did that, we'd sometimes break out of this loop one iteration earlier. Namely, in the case when we `borrow` and create a new connection, but think we didn't because `requireNew` was set to `false`, and then hit `EOF` here with `isFresh` set to `false`, then we'd loop again with `requireNew` set to `true`, and then jump out. Whereas if we knew we had a new connection, we'd hit the bail-out case immediately.